### ISSUE TITLE
delombok task should be re-run when a `lombok.config` file changes

### ISSUE THREAD

**Issue by msridhar** [freefair/gradle-plugins#604]
Thanks for the great plugins!  Here's a branch to reproduce my issue:

https://github.com/msridhar/tainting-example/tree/called-methods-lombok

Check out the branch and run `./gradlew clean compileJava --console=plain`.  I see the following output:

```
$ ./gradlew clean compileJava --console=plain
> Task :clean
> Task :createCheckerFrameworkManifest
> Task :delombok
> Task :generateEffectiveLombokConfig

> Task :compileJava
/Users/msridhar/git-repos/tainting-example/build/generated/sources/delombok/java/main/calledmethods/BuilderTest.java:8: warning: [type.anno.before.modifier] write type annotation @NonNull() immediately before type, after modifiers [private]
  private Integer y;
```

(The compile errors don't matter.)  Now, edit the `lombok.config` in the repo root directory and remove the second line, `lombok.addLombokGeneratedAnnotation = true`.  Then, running `./gradlew compileJava` I see the following output:
```$ ./gradlew compileJava --console=plain
> Task :createCheckerFrameworkManifest UP-TO-DATE
> Task :delombok UP-TO-DATE
> Task :generateEffectiveLombokConfig

> Task :compileJava
/Users/msridhar/git-repos/tainting-example/build/generated/sources/delombok/java/main/calledmethods/BuilderTest.java:8: warning: [type.anno.before.modifier] write type annotation @NonNull() immediately before type, after modifiers [private]
  private Integer y;
```
The `delombok` task is erroneously marked as up-to-date, and if I look at the generated sources under `build/generated/sources/delombok/java`, they still contain `@lombok.Generated` annotations.

To fix this issue, I think the `delombok` task should be re-run when a `lombok.config` file has been edited.

**Comment by msridhar** [comment]
I see there is already a commit for this issue; thanks for looking into it so quickly!  Can we close out the issue as fixed, then?  I can test once the fix makes it into a release.

**Comment by msridhar** [comment]
Confirmed fixed in 6.6.1, thanks!
